std/range/primitives.d

```diff
@@ -1987,6 +1993,177 @@ ElementType!R moveAt(R, I)(R r, I i) if (isIntegral!I)
 }
 
 /**
+Pops the first element from $(D r) and returns it.
+It combines $(D front) and $(D popFront) in one method.
+
+Params:
+    r = Forward range
+
+Returns:
+    Next element of the input range
+ */
+auto frontPop(Range)(ref Range r)
+if (isForwardRange!Range)
+in
+{
+    assert(!r.empty, "empty range passed to " ~ __PRETTY_FUNCTION__);
```

Comment:
An assertion without a message should be plenty. It's downright silly to have to put __PRETTY_FUNCTION__ in an assertion message. The `AssertError` already has that information. If the problem is that stacktraces don't show the function names, then that is what needs to be fixed rather than trying to work around it all over the place.